src/wasm.h

```diff
@@ -1619,6 +1619,18 @@ template<> struct hash<wasm::Address> {
     return std::hash<wasm::Address::address_t>()(a.addr);
   }
 };
-}
+
+template <> struct hash<wasm::Name> : public unary_function<wasm::Name, size_t> {
```

Comment:
Could we split this into its own change? There's a bunch of cases where we use `cashew::IString` instead of `Name` in unordered sets and the like because of this issue and we can fix those along with it, but we probably don't want those in this change.